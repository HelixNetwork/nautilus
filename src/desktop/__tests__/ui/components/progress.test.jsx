// import React from 'react';
// import { shallow } from 'enzyme';

// import ProgressComponent from 'ui/components/progress';

// const props = {
//     progress: 45,
//     title: 'Updated',
//     type: 'circle',
// };


// describe('Progress component', () => {
//     test('Render the component', () => {
//         const wrapper = shallow(<ProgressComponent {...props} />);

//         expect(wrapper).toMatchSnapshot();
//     });

    test('Correct progress', () => {
        const wrapper = shallow(<ProgressComponent {...props} />);

        const titleE2 = wrapper.find('Progress');
        console.log('titleE2  ', wrapper.debug());
        expect(titleE2.percent).toEqual('${props.progress}%');

        // expect(
        //     wrapper
        //         .find(type)
        //         .props().theme.symbol,
        // ).toEqual(`${props.progress}%`);
    });

 
    // test('without type ', () => {
    //     const mockProps = Object.assign({}, props, { type: null });
    //     const wrapper = shallow(<ProgressComponent {...mockProps} />);
    //     console.log('without type  ', wrapper.debug());

    //     expect(wrapper.hasClass('type')).toBeFalsy();
    // });

    // test('with type send', () => {
    //     const mockProps1 = Object.assign({}, props, { type: 'send', progress: 45 });
    //     const wrapper = shallow(<ProgressComponent {...mockProps1} />);

    //     console.log('with type send   ', wrapper.debug());


    //     console.log("mockProps in send type:   ", mockProps1 );
    //     expect(wrapper.hasClass('type')).toBeTruthy();
        
    //     // expect(wrapper.mockProps1.type).toEqual('send');
    //     // expect(wrapper.mockProps1.progress).toEqual(45);
    // });
 
    // test('With title', () => {
    //     const wrapper = shallow(<ProgressComponent {...props} />);
    //     const titleEl = wrapper.find('p');

    //     console.log('With title  ', wrapper.debug());

    //     expect(titleEl).toHaveLength(1);
    //     expect(titleEl.text()).toEqual(props.title);
    // });

    // test('Without title', () => {
    //     const mockProps = Object.assign({}, props, { title: null });
    //     const wrapper = shallow(<ProgressComponent {...mockProps} />);

    //     expect(wrapper.find('p')).toHaveLength(0);
    // });

// });
