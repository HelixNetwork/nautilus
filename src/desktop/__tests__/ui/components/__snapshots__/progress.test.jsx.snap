// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component Render the component 1`] = `
<div
  className="progress circle"
>
  <t
    percent={45}
    status="error"
    style={
      Object {
        "border": "none !important",
        "height": "250px",
        "left": "0px",
        "position": "absolute !important",
        "top": "100px",
        "width": "250px",
      }
    }
    theme={
      Object {
        "default": Object {
          "color": "#ebba53",
          "symbol": "45%",
          "trailColor": "#152353",
        },
      }
    }
    type="circle"
  />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <p>
    Updated
  </p>
</div>
`;
